### PR TITLE
[Merged by Bors] - feat: add mul_iff right and mul_iff_left for (AE)(Strongly)Measurable

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
Add the lemma that says that if `f` is measurable, then `f+g` is measurable iff `g` is measurable. Then add the same but with `g+f`, and the same lemmas about AEMeasurable, StronglyMeasurable and AEStronglyMeasurable.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by loefflerd** [comment]
Can you comment on why you're tagging these lemmas with `measurability`? I don't claim any expertise in how the `measurability` tactic works, but these lemmas don't seem (to my inexpert eye) as if they would be helpful for that tactic, given that it already has `Measurable.mul` etc at its disposal.

**Comment by LorenzoLuccioli** [comment]
> Can you comment on why you're tagging these lemmas with `measurability`? I don't claim any expertise in how the `measurability` tactic works, but these lemmas don't seem (to my inexpert eye) as if they would be helpful for that tactic, given that it already has `Measurable.mul` etc at its disposal.

Thank you for your feedback.
I don't know how `measurability` works internally, I added the tag because I was hoping that in this way `measurability` would be able to solve automatically situations like the following one:

```
example [CommGroup M] [MeasurableMul₂ M] [MeasurableInv M] (hf : Measurable f) (hfg : Measurable (f * g)) :
    Measurable g := by
  -- measurability -- doesn't work
  sorry
```

But it seems that even with the tag it doesn't manage to prove this. So I guess that I could remove them, if you agree.

**Review by RemyDegenne — COMMENTED** [review]
Can you please remove the measurability attributes? Other than that, it looks good to me.

File: Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean [hunk 1]
```diff
@@ -452,6 +452,17 @@ protected theorem div [Div β] [ContinuousDiv β] (hf : StronglyMeasurable f)
 #align measure_theory.strongly_measurable.div MeasureTheory.StronglyMeasurable.div
 #align measure_theory.strongly_measurable.sub MeasureTheory.StronglyMeasurable.sub
 
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean:455]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean [hunk 2]
```diff
@@ -452,6 +452,17 @@ protected theorem div [Div β] [ContinuousDiv β] (hf : StronglyMeasurable f)
...
+theorem mul_iff_right [CommGroup β] [TopologicalGroup β] (hf : StronglyMeasurable f) :
+    StronglyMeasurable (f * g) ↔ StronglyMeasurable g :=
+  ⟨fun h ↦ show g = f * g * f⁻¹ by simp only [mul_inv_cancel_comm] ▸ h.mul hf.inv,
+    fun h ↦ hf.mul h⟩
+
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean:461]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean [hunk 3]
```diff
@@ -1355,6 +1366,17 @@ protected theorem div [Group β] [TopologicalGroup β] (hf : AEStronglyMeasurabl
 #align measure_theory.ae_strongly_measurable.div MeasureTheory.AEStronglyMeasurable.div
 #align measure_theory.ae_strongly_measurable.sub MeasureTheory.AEStronglyMeasurable.sub
 
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean:1369]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean [hunk 4]
```diff
@@ -1355,6 +1366,17 @@ protected theorem div [Group β] [TopologicalGroup β] (hf : AEStronglyMeasurabl
...
+theorem mul_iff_right [CommGroup β] [TopologicalGroup β] (hf : AEStronglyMeasurable f μ) :
+    AEStronglyMeasurable (f * g) μ ↔ AEStronglyMeasurable g μ :=
+  ⟨fun h ↦ show g = f * g * f⁻¹ by simp only [mul_inv_cancel_comm] ▸ h.mul hf.inv,
+    fun h ↦ hf.mul h⟩
+
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.lean:1375]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Group/Arithmetic.lean [hunk 5]
```diff
@@ -520,6 +520,32 @@ theorem measurableEmbedding_inv [InvolutiveInv α] [MeasurableInv α] :
 
 end Inv
 
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Group/Arithmetic.lean:523]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Group/Arithmetic.lean [hunk 6]
```diff
@@ -520,6 +520,32 @@ theorem measurableEmbedding_inv [InvolutiveInv α] [MeasurableInv α] :
...
+theorem Measurable.mul_iff_right {G : Type*} [MeasurableSpace G] [MeasurableSpace α] [CommGroup G]
+    [MeasurableMul₂ G] [MeasurableInv G] {f g : α → G} (hf : Measurable f) :
+    Measurable (f * g) ↔ Measurable g :=
+  ⟨fun h ↦ show g = f * g * f⁻¹ by simp only [mul_inv_cancel_comm] ▸ h.mul hf.inv,
+    fun h ↦ hf.mul h⟩
+
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Group/Arithmetic.lean:530]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Group/Arithmetic.lean [hunk 7]
```diff
@@ -520,6 +520,32 @@ theorem measurableEmbedding_inv [InvolutiveInv α] [MeasurableInv α] :
...
+theorem AEMeasurable.mul_iff_right {G : Type*} [MeasurableSpace G] [MeasurableSpace α] [CommGroup G]
+    [MeasurableMul₂ G] [MeasurableInv G] {μ : Measure α} {f g : α → G} (hf : AEMeasurable f μ) :
+    AEMeasurable (f * g) μ ↔ AEMeasurable g μ :=
+  ⟨fun h ↦ show g = f * g * f⁻¹ by simp only [mul_inv_cancel_comm] ▸ h.mul hf.inv,
+    fun h ↦ hf.mul h⟩
+
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Group/Arithmetic.lean:537]:
```suggestion
@[to_additive]
```

File: Mathlib/MeasureTheory/Group/Arithmetic.lean [hunk 8]
```diff
@@ -520,6 +520,32 @@ theorem measurableEmbedding_inv [InvolutiveInv α] [MeasurableInv α] :
...
+theorem Measurable.mul_iff_left {G : Type*} [MeasurableSpace G] [MeasurableSpace α] [CommGroup G]
+    [MeasurableMul₂ G] [MeasurableInv G] {f g : α → G} (hf : Measurable f) :
+    Measurable (g * f) ↔ Measurable g :=
+  mul_comm g f ▸ Measurable.mul_iff_right hf
+
+@[to_additive (attr := measurability)]
```
Comment (LorenzoLuccioli) [inline Mathlib/MeasureTheory/Group/Arithmetic.lean:543]:
```suggestion
@[to_additive]
```

**Review by LorenzoLuccioli — COMMENTED** [review]
Removed `measurability` attributes

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**